### DISCUSSION TITLE
More block types (tables, files, images) in expandable blocks

### DISCUSSION THREAD

**Discussion by emmerich** [GitbookIO/community#16] [Feature Requests]
### What's your feature/idea?

I'd like to be able to put more block types inside expandable blocks.

### What's your desired outcome?

Some block types that would be useful that are not currently possible:

- Tables
- Files
- Images

### What's the impact of this for you?

- [ ] Critical
- [ ] High
- [ ] Medium
- [ ] Low

**Comment by ChucKN0risK** [comment]
I agree. I was also trying to add a Tab and child code blocks inside an expandable block.

**Reply by rodrcastro** [reply]
Hey! Thank you for your suggestion and engagement here :) I was curious to hear though on you not being able to add code blocks inside an expandable block. That should be doable:

<img width="715" alt="image" src="https://user-images.githubusercontent.com/18582098/213257179-cefaa677-8163-451d-bd6d-bcebcc6230d3.png">

Please try a [hard refresh](https://docs.gitbook.com/resources/faq/how-to-hard-refresh-your-browser) on your side and see if that helps. If it doesn't, shoot us an email support@gitbook.com so we can take a closer look at this :)

**Comment by taszap** [comment]
Another issue is linked to this #73

**Comment by aikoii** [comment]
Allow expandable blocks inside other expandable blocks was suggested here https://github.com/orgs/GitbookIO/discussions/164

**Comment by benzitney** [comment]
Agree with this. We are trying to condense a lot of articles with table content into a single article, and were planning to enclose the table content in expandables in order to avoid the article from getting ridiculously long. But since we can't include tables in expandables, this isn't an option

**Comment by srtena** [comment]
Pretty please 🙏
Not being able to add other blocks makes the expandable blocks practically useless. We've been planning to use them for tutorials, advanced explanations, and other cases, and every time we've discarded them because they don't support image resizing or other blocks, such as tables, inside.

**Comment by PlethoraChutney** [comment]
I really like expandable blocks! I write educational materials for complex topics, and it's nice to be able to hide advanced material in an expandable block so that new learners aren't overwhelmed by theory when they're first learning how to do something. However, the default inline image size (which is the only wan to include figure in expandable blocks right now, as far as I know) results in an unacceptably ugly block:

<img width="700" alt="image" src="https://github.com/GitbookIO/community/assets/27385979/1412118c-5d23-4dd8-be37-69fc0db354c8">

So at the very least, it would be extremely helpful if normal image blocks could be added inside expandable blocks, or even if inline images could be set to take the full width of the expandable block.

Thank you for considering this feature!

**Comment by Noikii** [comment]
Yes!! To all the comments above. I love the Expandable Tool but it would make a huge difference if you could size up pictures or add hints in there.

**Comment by denisemcg** [comment]
I agree with the comments above. I want to use more expandable blocks, but the content limitations are preventing me from doing so. I especially need to add hints, tables, and videos into expandable blocks. I have pages with many videos that are taking up too much space.

**Comment by PlethoraChutney** [comment]
Hello again! For our use case, I consider this critical functionality. I'm wondering if any devs/maintianers could comment on whether they think that allowing other block types (especially image blocks) inside expandable blocks is feasible and/or whether it's on the roadmap at all. I understand that my priorities do not necessarily reflect the broader user base of GitBook, but it would be nice to know whether this is a next month / next year / probably never situation so I can plan accordingly. Thank you!

**Comment by LerMerGerd** [comment]
I also agree with comments above. Coming from Confluence, I am now needing to figure a way to reformat a lot of our articles since images, hints, and videos cannot be added.
